src/charm.py

```diff
@@ -83,7 +73,7 @@ def __init__(self, *args: typing.Any) -> None:
         self.framework.observe(self.on.rotate_secret_key_action, self._on_rotate_secret_key_action)
         self.framework.observe(self.on.secret_storage_relation_changed, self._on_config_changed)
```

Comment:
This needs to be changed, it should map to a dedicated method

Reply:
Updated, thanks!